### PR TITLE
Update version history to be viewable everywhere and highlight current version

### PR DESCRIPTION
**Pull request by tim-dot-org**
This PR changes the version history feature so that it gives some more information to the viewer about which version they are viewing. It now displays in all modes including read-only (view only) and allows the user to switch to a different version or restore a version from a read-only view. The dialog has been updated to highlight the currently displayed version (before it was just alternate rows were highlighted) and to move the buttons around slightly (see ticket for UI spec).

## Links

- jira ticket: [LP-1746](https://codedotorg.atlassian.net/browse/LP-1746)

## Testing story

Manually tested locally, and passes all UI tests.

### CODE REVIEW

File: apps/style/common.scss [hunk 1]
```diff
@@ -987,10 +1003,6 @@ $workspace-header-button-margin: ($workspace-headers-height - $workspace-header-
   }
 }
 
-#versions-header {
```
Comment (tim-dot-org) [inline apps/style/common.scss:990]: hm, not sure why this file got autoformatted/wasn't already formatted, but functional changes are here

File: apps/style/common.scss [hunk 2]
```diff
@@ -1117,7 +1132,7 @@ $workspace-header-button-margin: ($workspace-headers-height - $workspace-header-
   td {
     vertical-align: middle;
   }
-  tr:nth-child(even){
```
Comment (tim-dot-org) [inline apps/style/common.scss:1120]: and here

File: apps/src/templates/VersionRow.jsx [hunk 3]
```diff
@@ -35,35 +37,44 @@ export default class VersionRow extends React.Component {
...
-        </button>
-      ];
+        </a>
+      );
     }
 
     return (
-      <tr className="versionRow">
+      <tr
+        className={[
```
Comment (maureensturgeon) [inline apps/src/templates/VersionRow.jsx]: Could use `classnames` library here with `classnames('versionRow', {'activeVersion': this.props.isActive})`

File: apps/src/templates/VersionHistory.jsx [hunk 4]
```diff
@@ -183,6 +184,11 @@ export default class VersionHistory extends React.Component {
               versionId={version.versionId}
               lastModified={new Date(version.lastModified)}
               isLatest={version.isLatest}
+              isActive={
```
Comment (maureensturgeon) [inline apps/src/templates/VersionHistory.jsx]: Nit: I was a little confused by what `isActive` means, it means the version the user is currently viewing right? I'm not sure what a more clear name might be. Perhaps `isCurrentlyViewed`, `isBeingViewed`?

File: apps/src/templates/VersionRow.jsx [hunk 5]
```diff
@@ -35,35 +37,44 @@ export default class VersionRow extends React.Component {
         </button>
       );
     } else {
-      button = [
+      buttons.push(
```
Comment (maureensturgeon) [inline apps/src/templates/VersionRow.jsx:40]: Is this covered by any tests?

File: apps/src/templates/VersionRow.jsx [hunk 6]
```diff
@@ -35,35 +37,44 @@ export default class VersionRow extends React.Component {
...
+          className="btn-info"
+          onClick={this.props.onChoose}
+        >
+          {msg.restoreThisVersion()}
+        </button>
+      );
+    }
+
+    if (!this.props.isActive) {
+      buttons.push(
```
Comment (maureensturgeon) [inline apps/src/templates/VersionRow.jsx:58]: Is this being covered by any tests?

File: apps/src/templates/VersionRow.jsx (hunk 5 above)
Comment (maureensturgeon) [inline apps/src/templates/VersionRow.jsx:44]: Is this hidden for users who don't own the code?

Comment (tim-dot-org) [reply apps/src/templates/VersionRow.jsx:44]: oh, good point. i think 'own the code' and 'read-only' gets used interchangably in the code? but this button should not be available to teachers once I make the change to let them see it.

Comment (tim-dot-org) [reply apps/src/templates/VersionRow.jsx:44]: used project.isOwner to switch this button off

**Review by maureensturgeon — APPROVED** [review]

**Pull request merged**